feat(docs): add performance bench and tuning pages (#2689)
kestra-io/docs: content/docs/performance/performance-tuning.md

---
title: Performance Tuning
icon: TODO
---

> Not all workloads are the same, so Kestra is configured to balance throughput (the ability to process a lot of executions in parallel) and latency (the ability to process executions quickly) without using too much resource.

Reviewer: ```suggestion
Not all workloads are the same, so Kestra is configured to balance throughput (the ability to process a lot of executions in parallel) and latency (the ability to process executions quickly) without using too many resources.
```

> On this page, we will introduce some performance tuning that may be done to improve the performance of the Kestra orchestrator, all comes with tradeoff so be sure to understand them before applying them.

Reviewer: ```suggestion
On this page, we will introduce some performance tuning that may be done to improve the performance of the Kestra orchestrator, all of which come with tradeoffs, so be sure to understand them before applying them.
```

> Before reading further, be sure to understand the [Kestra architecture](../07.architecture).

Reviewer: ```suggestion
Before you read more, please be sure to understand the [Kestra architecture](../07.architecture).
```

## The Worker

> The Worker is responsible for executing your tasks, so the way to tune it is dependent on what you execute.

Reviewer: ```suggestion
The Worker is responsible for executing your tasks, so how to tune it depends on what you execute.
```

> The most useful configuration is the number of Worker threads, this is by default 4 times the number of available cores.

Reviewer: ```suggestion
The most useful configuration is the number of Worker threads, which is 4 times the number of available cores by default.
```

You can increase it to increase parallelism task execution, depending on how you start Kestra, this would be:

> - If using the standalone: setting `--worker-thread` in the standalone command line.

Reviewer: ```suggestion
- If using the standalone, set `--worker-thread` in the standalone command line.
```

> - If using separate component processes: setting `--thread` in the Worker command line.

Reviewer: ```suggestion
- If using separate component processes, set `--thread` in the Worker command line.
```

> - If using our Helm chart, setting `deployments.worker.workerThreads` in the values.

Reviewer: ```suggestion
- If using our Helm chart, set `deployments.worker.workerThreads` in the values.
```

## The JDBC backend

### The JDBC queue

> The JDBC queue is the most performant-sensitive component of the JDBC backend.

Reviewer: ```suggestion
The JDBC queue is the most performance-sensitive component of the JDBC backend.
```

It may be configured using the following configuration options:

```yaml
kestra:
  jdbc:
    queues:
      minPollInterval: 25ms
      maxPollInterval: 500ms
      pollSwitchInterval: 60s
      pollSize: 100
```

> How does it works? The JDBC queue will poll a database table for new messages; this poll is made each 25ms by default, with a limit of 100 records.

Reviewer: ```suggestion
How does it work? The JDBC queue polls a database table for new messages; this poll is made every 25ms by default and has a limit of 100 records.
```
